### PR TITLE
Fix past appointment check when VSP flag is on

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
Fix past appointment check when VSP flag is on #11657

## Testing done
Unit testing

## Screenshots
N/A

## Acceptance criteria
- [ ] All unit test should pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/utils/data.js [hunk 1]
```diff
@@ -44,6 +45,29 @@ function getUserMessage(data) {
...
+// When useVSP is true, the clinic data is using the FHIR format. The clinic id format is:
+// urn:va:healthcareservice:983:983:308
+// where the last 3 tokens represent the site id, facility id, and clinic id.
+function getToken(clinic, index) {
+  return clinic.identifier[0].value.split(':')[index];
+}
+
+function getSiteCode(clinic, useVSP) {
+  if (clinic) {
+    return useVSP
```
Comment (jbalboni) [inline src/applications/vaos/utils/data.js:57]: Our transformer for HealthcareService creates those URN identifiers: https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/vaos/services/healthcare-service/transformers.js#L20

So I think you could skip the feature toggle check entirely in this PR and always pull the ids you need with this `getToken` call.

File: src/applications/vaos/utils/data.js [hunk 2]
```diff
@@ -44,6 +45,29 @@ function getUserMessage(data) {
   return `${label}: ${data.reasonAdditionalInfo}`;
 }
 
+// When useVSP is true, the clinic data is using the FHIR format. The clinic id format is:
+// urn:va:healthcareservice:983:983:308
+// where the last 3 tokens represent the site id, facility id, and clinic id.
+function getToken(clinic, index) {
```
Comment (jbalboni) [inline src/applications/vaos/utils/data.js:51]: `getToken` is a pretty generic name, what about something like `getURNToken` or `getIdentifierToken`?

**Review by jbalboni — COMMENTED** [review]
Looks good, just a couple minor comments

File: src/applications/vaos/utils/data.js (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/utils/data.js]: Done.

File: src/applications/vaos/utils/data.js (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/utils/data.js]: Done.

File: src/applications/vaos/services/healthcare-service/transformers.js [hunk 3]
```diff
@@ -159,7 +159,36 @@ export function findCharacteristic(clinic, characteristic) {
     return element.text === characteristic;
   });
 
-  return result.coding.code && result.coding.code !== undefined
-    ? result.coding.code
-    : result.coding.display;
+  if (result) {
```
Comment (jbalboni) [inline src/applications/vaos/services/healthcare-service/transformers.js:162]: I think this block could be `return result?.coding?.code || result?.coding?.display;`

**Review by jbalboni — APPROVED** [review]
This looks good to me. I found in testing on the review instance that there's a similar check to the one you fixed in the reducer for the clinics page, we can handle that in a separate ticket.

File: src/applications/vaos/services/healthcare-service/transformers.js (hunk 3 above)
Comment (vbahinwillit) [reply src/applications/vaos/services/healthcare-service/transformers.js:162]: Done

**Pull request merged**